numba/cuda/decorators.py

```diff
@@ -145,14 +137,6 @@ def device_jit(func):
             return kernel_jit
 
 
-def declare_device(name, restype=None, argtypes=None):
-    restype, argtypes = convert_types(restype, argtypes)
+def declare_device(name, sig):
+    argtypes, restype = sigutils.normalize_signature(sig)
```

Comment:
How was this working?!? https://github.com/numba/numba/blob/3b1e4abd1b82ab5dc60f912ad43faa6e26cd87ba/numba/cuda/tests/cudadrv/test_linker.py#L72

Reply:
`restype` was misleading - it could have been a signature as well, because of what `convert_types` did with it (see just below).

I have spent a fair bit of confused time looking at this already! :-)

Reply:
Thanks :) was also confused!